bin/mail/2016-09-12-back-to-school/generate_list_functions.rb

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require 'active_support'
+require 'active_support/core_ext/object/blank'
+
+ALL_EMAILS = Set.new
+
+def dedupe(contacts)
+  contacts.except! *ALL_EMAILS
+  ALL_EMAILS.merge contacts.keys
```

Comment:
You would think, but apparently not with [Set](http://ruby-doc.org/stdlib-2.3.1/libdoc/set/rdoc/Set.html#method-i-merge)